src/slowmokit/methods/neighbors/categorical_nb/categorical_nb.cpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file methods/neighbors/gaussian_nb/categorical_nb.cpp
+ *
+ * Implementation of the Categorical Naive Bayes main program
+ */
+
+#include "categorical_nb.hpp"
+
+template<class T>
+
+std::string fitPredict(std::vector<std::vector<T>> xTrain,
+                       std::vector<std::string> yTrain, std::vector<T> xTest)
+{
+  // posterior = (prior * likelihood)/evidence
+  // since, evidence is same among all instances -> we can ignore it
+
+  if (xTrain.size() == 0 || yTrain.size() == 0)
+    throw "Make sure that you have atleast one train example";
+  if (xTrain.size() != yTrain.size())
+    throw "Number of features and target must be equal";
+
+  std::map<std::string, double> priors;
+  std::map<std::string, int> occurences;
+
+  for (auto category : yTrain)
+  {
+    occurences[category]++;
+  }
+  for (auto current : occurences)
+  {
+    priors[current.first] = double(current.second) / yTrain.size();
+  }
+  // priors calculated
+
+  std::map<std::string, std::map<T, double>> likelihoods;
+  // for any ith label -> n instances -> for each differ probability
```

Comment:
reframe comment in normal english. No shorthand notations.

Reply:
reframed , but i have made some changes in code